standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -548,6 +548,16 @@ public void addPartitions(List<MPartition> parts, List<List<MPartitionPrivilege>
    */
   public List<Partition> alterPartitions(MTable table, List<String> partNames,
                                          List<Partition> newParts, String queryWriteIdList) throws MetaException {
+
+    for (Partition tmpPart : newParts) {
```

Comment:
Can we move this common check into `ObjectStore#alterPartitions`?

Reply:
done